src/plugins/intel_gpu/src/graph/graph_optimizer/mark_shape_of_subgraphs.cpp

```diff
@@ -15,12 +16,116 @@
+        for (auto& reorder_dep : reorder_deps) {
+            // broadcast node
+            if (reorder_dep.first->is_type<broadcast>()) {
+                auto& broadcast_deps = reorder_dep.first->get_dependencies();
+                return has_shape_of_dep(broadcast_deps);
+            }
+        }
+        return false;
+    }
+
+    bool has_reorder_reoder_dep(const std::vector<std::pair<cldnn::program_node*, int>>& eltwise_deps) {
+        for (auto& eltwise_dep : eltwise_deps) {
+            // reorder node (reorder -> eltwise)
+            if (eltwise_dep.first->is_type<reorder>()) {
+                auto& eltwise_dep_reorder_deps = eltwise_dep.first->get_dependencies();
+
+                for (auto& eltwise_dep_reorder_dep : eltwise_dep_reorder_deps) {
+                    // reorder node (broadcast -> reorder)
+                    if (eltwise_dep_reorder_dep.first->is_type<reorder>()) {
+                        auto& reorder_dep_reorder_deps = eltwise_dep_reorder_dep.first->get_dependencies();
+                        return has_broadcast_dep(reorder_dep_reorder_deps);
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    bool has_eltwise_dep(const std::vector<std::pair<cldnn::program_node*, int>>& reorder_deps) {
+        for (auto& reorder_dep : reorder_deps) {
+            // eltwise node
+            if (reorder_dep.first->is_type<eltwise>()) {
+                auto& eltwise_deps = reorder_dep.first->get_dependencies();
+                return has_reorder_reoder_dep(eltwise_deps);
+            }
+        }
+        return false;
+    }
+
+    bool has_reorder_dep(const std::vector<std::pair<cldnn::program_node*, int>>& conv_deps) {
+        for (auto& conv_dep : conv_deps) {
+            //if (conv_dep.first->id().find(dequantize_name) != std::string::npos) {
+
+            // reorder node ( reorder -> convolution)
+            if (conv_dep.first->is_type<reorder>()) {
+                auto& reorder_deps = conv_dep.first->get_dependencies();
+                return has_eltwise_dep(reorder_deps);
+            }
+        }
+        return false;
+    }
+
+    bool has_convolution_dep(const std::vector<std::pair<cldnn::program_node*, int>>& dependencies) {
+        for (auto& dependency : dependencies) {
+            // convolution node
+            if (dependency.first->is_type<convolution>()) {
+                auto& conv_deps = dependency.first->get_dependencies();
+                return has_reorder_dep(conv_deps);
+            }
+        }
+        return false;
+    }
+
+    // check dependencies for reorder node added for convolution in quantized model
+    bool skip_quantization_conv_reorder(const program_node& node) {
+        // reorder -> convolution -> reorder -> eltwise -> reorder -> reorder -> broadcast -> shape_of -> input_layout
+        if (!node.is_type<reorder>()) {
+            return false;
+        }
```

Comment:
I think the expected primitives sequence is too narrow for a single particular case. Could you try to exclude all Broadcasts with ShapeOf dependency (when broadcast operation involves broadcasting a constant to some specific shape), and check if this somehow affects other models?

Reply:
@sshlyapn There was a commit [Change to skip mark_node in shape flow for broadcast node if dependency node is shape_of](https://github.com/openvinotoolkit/openvino/pull/27786/commits/291e61d63b323f3f3e76b31d7c827b79be03be86) in this PR.

And a performance drop was observed with that commit (chatglm3 & qwen for example):
![image](https://github.com/user-attachments/assets/d8fec115-194e-4bfd-9816-48d4cb520c4b)

Reply:
@yuanxion did you have a chance to check the inputs order of chatglm3 and qwen model cases? There might be the difference between `Broadcast(const, ShapeOf)` vs `Broadcast(ShapeOf, const)`

Reply:
@sshlyapn Only PixArt model has the `Constant + ShapeOf -> Broadcast` pattern.

- For the PixArt-alpha/PixArt-LCM-XL-2-1024-MS model, the dependency order of broadcast is Broadcast(Constant, ShapeOf):

```
// Constant + ShapeOf -> Broadcast
node: broadcast:hidden_states/fq_output_0	
dep [0]: constant:Constant_36836
dep [1]: shapeof:ShapeOf_36839
```

- For llama-3-8b, the order is Broadcast(Select, ShapeOf)

```
// Select + ShapeOf -> Broadcast
node: broadcast:__module.model/aten::copy_/Broadcast			
dep [0]: select:__module.model/aten::masked_fill/Select
dep [1]: shapeof:__module.model/aten::copy_/ShapeOf
```
![image](https://github.com/user-attachments/assets/688ce875-3189-4f7c-a6cc-80a471c831d1)

- And didn't see other kinds of pattern like `ShapeOf -> Broadcast` in qwen-7b-chat/llama-3-8b/chatglm3-6b/gemma-7b-it. `ShapeOf -> Gather` is often seen instead in these models.